AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_entity_utils.py

```diff
@@ -470,3 +470,99 @@ def focus_on_owning_prefab(self) -> None:
         assert self.id.isValid(), "A valid entity id is required to focus on its owning prefab."
         focus_prefab_result = azlmbr.prefab.PrefabFocusPublicRequestBus(bus.Broadcast, "FocusOnOwningPrefab", self.id)
         assert focus_prefab_result.IsSuccess(), f"Prefab operation 'FocusOnOwningPrefab' failed. Error: {focus_prefab_result.GetError()}"
+
+
+class EditorLevel:
+    """
+    EditorLevel class used to add and fetch level components.
+    Level entity is a special entity that you do not create/destroy independently of larger systems of level creation.
+    This collects a number of staticmethods that do not rely on entityId since Level entity is found internally by
+    EditorLevelComponentAPIBus requests.
+    """
+
+    @staticmethod
+    def get_type_ids(component_names: list) -> list:
+        """
+        Used to get type ids of given components list for EntityType Level
+        :param: component_names: List of components to get type ids
+        :return: List of type ids of given components.
+        """
+        type_ids = editor.EditorComponentAPIBus(
+            bus.Broadcast, "FindComponentTypeIdsByEntityType", component_names, azlmbr.entity.EntityType().Level
+        )
+        return type_ids
+
+    @staticmethod
+    def add_component(component_name: str) -> EditorComponent:
+        """
+        Used to add new component to Level.
+        :param component_name: String of component name to add.
+        :return: Component object of newly added component.
+        """
+        component = EditorLevel.add_components([component_name])[0]
+        return component
+
+    @staticmethod
+    def add_components(component_names: list) -> List[EditorComponent]:
```

Comment:
nit: these are nearly identical to the methods in the EditorEntity class. Did you consider implementing this class as a wrapper around the existing EditorEntity class rather than using the EditorLevelComponentAPIBus?

Using something along the lines of 

    def __init__(self):
        level_entity_id = azlmbr.editor.ToolsApplicationRequestBus(azlmbr.bus.Broadcast, 'GetCurrentLevelEntityId')
        self.editor_entity: EditorEntity(level_entity_id)
        
then a function like this would just be
    def add_components(component_names: list) -> List[EditorComponent]:
        self.editor_entity.add_components(list)

Reply:
I considered it as a good option, but the existing methods rely on EditorComponent.get_type_ids() which is hardcoded to azlmbr.entity.EntityType().Game. One option was to address that was to add a parameter for type with a default to .Game. Then all methods would have to bubble up optional param for level versions to switch the type to .Level. I also had some concerns about overloading the Game component entity bus. With a dedicated bus for level components existing, if any change to the systems for level components were to occur it might not reflect through the game component bus. right now we could probably get away with this, but then we would be open to unexpected failure later if this changed. It also might be messy under the hood with optional parameters being hidden loaded into the level component calls.